website/main/content/blog/2025-06-30-sector-expiration.md

```diff
@@ -0,0 +1,213 @@
+3.  **Load-Balanced Archiving (New History):** As new history arrives, it gets stored across
+    farmers/plots without large re-plotting events.
+4.  **Detect Expiration and Validity:** Plots, sectors and blocks include all the information needed
+    to verify their validity, plot assignment to a shard, sector belonging to the plot, and
+    expiration.
+5.  **No Excessive Re-plotting:** Keep re-plotting burden similar to original Subspace.
+
+### Committing sectors to history ranges
+
+The core idea for sector expiration is the following: instead of letting farmers commit their
+sectors to any history size, they are bound to commit them to a range of history sizes determined by
+the plot the sector is linked to. This way, we limit the range of history sizes that a sector can be
+committed to, and hence the parallel sectors with the same `plot_id` that and different committed
+history sizes that can be created. This is implemented as a "self-evolving plot commitment range"
+that grows as the blockchain grows on top of the existing mechanism for sector expiration. Thus, we
+maintain all the properties of the original Subspace protocol, while adapting it to the needs of the
+sharded version. This allows us to also learn an apply the heuristics from the live Subspace
+network.
+
+The high-level operation of the protocol is as follows:
+
+- When creating a new plot, farmers need to choose a random offset (essentially a nonce) for their
+  plot. This offset determines the specific ID of the plot and will influence the history range that
+  the plot's sectors can commit to. As a reminder, plots identities are inferred from the public key
+  hash of the farmer, and the random offset they choose:
+
+```rust
+plot_id = hash(public_key_hash || offset)
+```
+
+- When creating new sectors for a plot, farmers will choose any valid history size within the plot's
+  current range. Piece selection is based on the sector's specific history size, and expiration
+  follows the same logic as the original Subspace protocol. Finally, the sector ID is derived from
+  the plot ID, sector index, and committed history size:
+
+```rust
+sector_id = keyed_hash(plot_id, sector_index || committed_history_size)
+```
+
+- We determine a protocol parameter `BASE_WINDOW_SIZE` that determines the size of the `BASE_RANGE`.
+  This `BASE_RANGE` determines the baseline history range for all plots. This `BASE_RANGE`
+  determines the size of history sizes that plots can commit to initially. The plot offset modifies
+  the `BASE_RANGE` to create a unique range for each plot ID, and as the history of the chain grows,
+  this is expanded accordingly so it can fit new history sizes. The following image belongs to a
+  [simple visualisation tool](./2025-06-30-sector-expiration/range-evolution-visual.html) that an
+  LLM has built for me to visualise how the base range and effective range of a plot behaves as the
+  history of the chain grows:
```

Comment:
I know you tried, but I wouldn't be able to understand anything from such description :sweat_smile: